blocks/editable/content.js

```diff
@@ -0,0 +1,117 @@
+import { renderToString } from '@wordpress/element';
+import { last } from 'lodash';
+import { childrenToReact } from './tree';
+
+export function nodeListToReact( editor, nodeList ) {
+	const fragment = editor.getDoc().createDocumentFragment();
+
+	nodeList.forEach( function( node ) {
+		fragment.appendChild( node.cloneNode( true ) );
+	} );
+
+	return childrenToReact( editor.serializer.serialize( fragment, { format: 'tree' } ) );
+}
+
+function splitResult( before, after ) {
+	return { before: before, after: after };
+}
+
+export function setContent( editor, content ) {
+	if ( ! content ) {
+		content = '';
+	}
+
+	content = renderToString( content );
+	editor.setContent( content, { format: 'raw' } );
+}
+
+export function getContent( editor ) {
+	return childrenToReact( editor.getContent( { format: 'tree' } ) );
```

Comment:
What shape has this "tree" format? Can you share an example?

Do you think it makes sense to avoid the "react" element format at all and just use the tree format in the state as well (outside Editable), in which case, it requires some refactoring but might be worth it (same as #3048)

Reply:
I created a simple gist doc on the structure of this tree.
https://gist.github.com/spocke/b009c5ef460c188c8995eb46b064a6b7

I guess using it for tinymce related things makes sense but unsure if it's a general enough format for usage for other things since each node has methods on it for mutation the react format at least it's input seems simpler in that regard.

Reply:
Can you serialize this "tree" using `JSON.stringify` and recreate later after calling `JSON.parse`? The root cause why we don't want to store React objects in the Redux state is that rehydration doesn't work out of the box.

Reply:
We could convert the tree into the simple json format that was suggested else where. It's pretty fast to just convert a one tree structure to another if it's just simple objects. We just need to implement a way to set the tree back in tiny just before the filters apply.